test/python/pulse/compiler_passes/test_set_sequence.py

```diff
@@ -0,0 +1,379 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test SetSequence"""
+from test import QiskitTestCase
+from ddt import ddt, named_data, unpack
+
+from qiskit.pulse import (
+    Constant,
+    Play,
+    Delay,
+    ShiftPhase,
+)
+
+from qiskit.pulse.ir import (
+    SequenceIR,
+)
+
+from qiskit.pulse.model import QubitFrame, Qubit, MixedFrame
+from qiskit.pulse.transforms import (
+    AlignLeft,
+    AlignRight,
+    AlignSequential,
+    AlignFunc,
+    AlignEquispaced,
+)
+from qiskit.pulse.compiler import MapMixedFrame, SetSequence
+from qiskit.pulse.exceptions import PulseError
+from .utils import PulseIrTranspiler
+
+
+@ddt
+class TestSetSequenceParallelAlignment(QiskitTestCase):
+    """Test SetSequence pass with Parallel Alignment"""
+
+    ddt_named_data = [["align_left", AlignLeft()], ["align_right", AlignRight()]]
+
+    def _get_pm(self) -> PulseIrTranspiler:
+        pm = PulseIrTranspiler()
+        pm.append(MapMixedFrame())
+        pm.append(SetSequence())
+        return pm
+
+    @named_data(*ddt_named_data)
+    @unpack
+    def test_no_mapping_pass_error(self, alignment):
+        """test that running without MapMixedFrame pass raises a PulseError"""
+
+        pm = PulseIrTranspiler()
+        pm.append(SetSequence())
+        ir_example = SequenceIR(alignment)
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+
+        with self.assertRaises(PulseError):
+            pm.run(ir_example)
+
+    @named_data(*ddt_named_data)
+    @unpack
+    def test_single_instruction(self, alignment):
+        """test with a single instruction"""
+
+        ir_example = SequenceIR(alignment)
+        ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+
+        ir_example = self._get_pm().run(ir_example)
+        edge_list = ir_example.sequence.edge_list()
+        self.assertEqual(len(edge_list), 2)
+        self.assertTrue((0, 2) in edge_list)
+        self.assertTrue((2, 1) in edge_list)
+
+    # TODO: Take care of this weird edge case
+    # def test_instruction_not_in_mapping(self):
+    #     """test with an instruction which is not in the mapping"""
+    #
+    #     ir_example = SequenceIR(AlignLeft())
+    #     ir_example.append(Play(Constant(100, 0.1), mixed_frame=MixedFrame(Qubit(0), QubitFrame(1))))
+    #     ir_example.append(Delay(100, target=Qubit(5)))
```

Comment:
Oh this is an interesting edge case. Probably this can be addressed with a resource mapping data. Let's leave this test, but not with comments. Please use https://docs.python.org/3/library/unittest.html#skipping-tests-and-expected-failures